rflx/pyrflx/typevalue.py

```diff
@@ -251,23 +252,64 @@ def literals(self) -> Mapping[Name, Expr]:
         return {Variable(k): v for k, v in self._type.literals.items()}
 
 
-class OpaqueValue(TypeValue):
+class CompositeValue(TypeValue):
+
+    _expected_size: Optional[Expr]
+
+    def __init__(self, vtype: Composite) -> None:
+        self._expected_size = None
+        super().__init__(vtype)
+
+    def set_expected_size(self, expected_size: Expr) -> None:
+        self._expected_size = expected_size
+
+    def _check_length_of_assigned_value(
+        self, value: Union[bytes, Bitstring, List[TypeValue]]
+    ) -> None:
+        if isinstance(value, bytes):
+            length_of_value = len(value) * 8
+        elif isinstance(value, Bitstring):
+            length_of_value = len(value)
+        else:
+            bits = [str(element.bitstring) for element in value]
+            length_of_value = len(Bitstring(str.join("", bits)))
```

Comment:
See my other comment about implementing `join` for `Bitstring`.

Reply:
Fixed by 7f48161